### PR TITLE
Lopez_143

### PR DESCRIPTION
**Pull request by jlopez81**

### CODE REVIEW

File: CHANGES_README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Software Repository for the IEEE Mars Rover project
+
+Changes done to the RoverSystem code: 
+
+1. Created a Logger folder in the src folder with the file logger.cpp and logger.hpp. Also created a Makefile to compile and create a library
+2. Created a LibLogger.a so that the logger functions can be used anywhere. 
+3. In the rover main.cpp replaced any printfs or couts with the log function. Any code with a exit(1) or return 1 had the Level ERROR and 
+any warnings we given the level WARNING, anything else was given the level INFO. 
+
+-Josue Lopez
```
Comment (layneson) [inline CHANGES_README.md:11]: Thanks for the description of changes! It is really appreciated. However, putting it in a file like this would result in a bunch of files sitting around that aren't well-tied to any particular task. In the future, put these kinds of statements in your commit messages and/or the body of the pull request. For example, in the commit where you create the logger folder and move stuff there, you could put (1) above as the commit message. Don't worry about it for this one; I'll remove this file. If you have any Git questions, feel free to ask!

**Review by layneson — CHANGES_REQUESTED** [review]
You committed a bunch of binary files (*.a, bin/rover, etc). Anything in the /bin directory should not be in Git, as those output files are unique to your architecture (and it's why the bin directory is in the .gitignore file). Unfortunately, even if you remove them now, they were committed at some point so they will always exist in the repository history, but don't worry about that... I will fix it. Also, it looks like from your commit messages that you were using the Github web interface, which can make using Git properly very difficult. Take a look at the Github desktop app, which can make Git a lot easier if you don't want to use the command line. Otherwise, I can give a mini tutorial or any pointers that may help!

Another thing: now that we have the logger library extracted, the base station should refer to that external library now (in the same way that the rover does). However, in your branch, the base station still uses logger.hpp & logger.cpp inside its source. This is an issue because if we make changes to the logging library, the base station won't get the updates (which defeats the purpose of a shared library in the first place). So another thing you need to do is delete the logger files from the base station and modify the makefile to use the new logging library (in the same manner as the rover).

Also: I ran the rover in your branch and I didn't see any output to the console. When I looked further, I realized that you did not create a logger handler on the rover side (handlers are how the log messages get anywhere). Put this in rover/main.cpp above the main function:
```
void stderr_handler(logger::Level level, std::string message) {
    fprintf(stderr, message.c_str());
}
```
That defines a handler that just prints everything to standard error. Then put this at the top of the main function:
```
logger::register_handler(stderr_handler);
```
That will register the handler with the logger library and when you build and run the rover, you should see some output regarding cameras appear in the console.

To summarize, please make the following changes:
1. Don't commit anything in the bin/ directory in the future.
2. Delete the logger files in src/base_station and edit the base station Makefile to use the new logger library.
3. Add the stderr handler described above to the rover.

Everything else looks great and your code compiles. Thanks for your work and I look forward to reviewing the above changes! Let me know on slack when you are finished fixing the above. I _highly_ recommend learning and using Git via the command-line or, at the very least, the GitHub desktop app.
